test/src/test/java/jenkins/cli/StopBuildsCommandTest.java

```diff
@@ -0,0 +1,217 @@
+        waitForLastBuildToStop(project);
+
+        final String out = runWith(Collections.singletonList(TEST_JOB_NAME)).stdout();
+
+        assertThat(out, equalTo("No builds stopped" + LN));
+    }
+
+    @Test
+    public void shouldStopSeveralWorkingBuilds() throws Exception {
+        final FreeStyleProject project = createLongRunningProject(TEST_JOB_NAME);
+        project.setConcurrentBuild(true);
+
+        setupSlaveWithTwoExecutors();
+
+        project.scheduleBuild2(0).waitForStart();
+        project.scheduleBuild2(0).waitForStart();
+
+        final String stdout = runWith(Collections.singletonList(TEST_JOB_NAME)).stdout();
+
+        assertThat(stdout, equalTo("Build '#2' stopped for job 'jobName'" + LN +
+                "Build '#1' stopped for job 'jobName'" + LN));
+        waitForLastBuildToStop(project);
+    }
+
+    @Test
+    public void shouldReportNotSupportedType() throws Exception {
+        final String testFolderName = "folder";
+        j.createFolder(testFolderName);
+
+        final String stderr = runWith(Collections.singletonList(testFolderName)).stderr();
+
+        assertThat(stderr, equalTo(LN + "ERROR: Job not found: 'folder'" + LN));
+    }
+
+    @Test
+    public void shouldDoNothingIfJobNotFound() throws Exception {
+        final String stderr = runWith(Collections.singletonList(TEST_JOB_NAME)).stderr();
+
+        assertThat(stderr, equalTo(LN + "ERROR: Job not found: 'jobName'" + LN));
+    }
+
+    @Test
+    public void shouldStopWorkingBuildsInSeveralJobs() throws Exception {
+        final List<String> inputJobNames = asList(TEST_JOB_NAME, TEST_JOB_NAME_2);
+        setupAndAssertTwoBuildsStop(inputJobNames);
+    }
+
+    @Test
+    public void shouldFilterJobDuplicatesInInput() throws Exception {
+        final List<String> inputNames = asList(TEST_JOB_NAME, TEST_JOB_NAME, TEST_JOB_NAME_2);
+        setupAndAssertTwoBuildsStop(inputNames);
+    }
+
+    @Test
+    public void shouldReportBuildStopError() throws Exception {
+        final FreeStyleProject project = createLongRunningProject(TEST_JOB_NAME);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().
+                grant(Jenkins.READ).everywhere().toEveryone().
+                grant(Item.READ).onItems(project).toEveryone().
+                grant(Item.CANCEL).onItems(project).toAuthenticated());
+        project.scheduleBuild2(0).waitForStart();
+
+        final String stdout = runWith(Collections.singletonList(TEST_JOB_NAME)).stdout();
+
+        assertThat(stdout,
+                equalTo("Exception occurred while trying to stop build '#1' for job 'jobName'. " +
+                        "Exception class: AccessDeniedException2, message: anonymous is missing the Job/Cancel permission" + LN +
+                        "No builds stopped" + LN));
+    }
+
+    @Test
+    public void shouldStopSecondJobEvenIfFirstStopFailed() throws Exception {
```

Comment:
Mm, probably not. You should not be running this command against jobs you do not have permission to abort and be expecting anything to work.

Reply:
Let's discuss [here](https://github.com/jenkinsci/jenkins/pull/3686#discussion_r281018439)